### PR TITLE
[fix](ngram bloomfilter) fix narrow conversion for ngram bf_size

### PR DESCRIPTION
**Pull request by airborne12**
### What problem does this PR solve?

Problem Summary:
Fix ngram bloomfilter index coredump as below
```
*** SIGFPE integer divide by zero (@0x56294f026472) received by PID 4016941 (TID 4019213 OR 0x7f294eb4d640) from PID 1325556850; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at be/src/common/signal_handler.h:421
 1# PosixSignals::chained_handler(int, siginfo*, void*) [clone .part.0] in bin/jdk-17.0.2/lib/server/libjvm.so
 2# JVM_handle_linux_signal in bin/jdk-17.0.2/lib/server/libjvm.so
 3# 0x00007F3071042520 in /lib/x86_64-linux-gnu/libc.so.6
 4# doris::segment_v2::NGramBloomFilter::add_bytes(char const*, unsigned int) at be/src/olap/rowset/segment_v2/ngram_bloom_filter.cpp:61
 5# doris::ITokenExtractorHelper<doris::NgramTokenExtractor>::string_to_bloom_filter(char const*, unsigned long, doris::segment_v2::BloomFilter&) const at be/src/olap/itoken_extractor.h:61
 6# doris::segment_v2::NGramBloomFilterIndexWriterImpl::add_values(void const*, unsigned long) at be/src/olap/rowset/segment_v2/bloom_filter_index_writer.cpp:250
```

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [x] Confirm test cases
- [ ] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_writer.cpp [hunk 1]
```diff
@@ -192,8 +192,19 @@ Status SegmentWriter::_create_column_writer(uint32_t cid, const TabletColumn& co
     if (tablet_index) {
         opts.need_bloom_filter = true;
         opts.is_ngram_bf_index = true;
-        opts.gram_size = tablet_index->get_gram_size();
-        opts.gram_bf_size = tablet_index->get_gram_bf_size();
+        //narrow convert from int32_t to uint8_t and uint16_t which is dangerous
+        auto gram_size = tablet_index->get_gram_size();
+        auto gram_bf_size = tablet_index->get_gram_bf_size();
+        if (gram_size > 256 || gram_size < 1) {
+            return Status::NotSupported("Do not support ngram bloom filter for ngram_size: ",
```
Comment (eldenmoon) [inline be/src/olap/rowset/segment_v2/segment_writer.cpp:199]: why not check gram in FE

File: fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java [hunk 2]
```diff
@@ -254,7 +254,7 @@ public void checkColumn(Column column, KeysType keysType, boolean enableUniqueKe
                     if (ngramSize > 256 || ngramSize < 1) {
                         throw new AnalysisException("gram_size should be integer and less than 256");
                     }
-                    if (bfSize > 65536 || bfSize < 64) {
+                    if (bfSize > 65535 || bfSize < 64) {
                         throw new AnalysisException("bf_size should be integer and between 64 and 65536");
```
Comment (eldenmoon) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java:258]: error message `bf_size should be integer and between 64 and 65535`

File: fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java [hunk 3]
```diff
@@ -254,7 +254,7 @@ public void checkColumn(Column column, KeysType keysType, boolean enableUniqueKe
...
                     }
                 } catch (NumberFormatException e) {
```
Comment (eldenmoon) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java:260]: maybe we shoud  add regression test for bfSize > 65535

**Comment by airborne12** [comment]
run buildall

**Review by gavinchou — DISMISSED** [review]

File: be/src/olap/rowset/segment_v2/segment_writer.cpp [hunk 4]
```diff
@@ -192,8 +192,19 @@ Status SegmentWriter::_create_column_writer(uint32_t cid, const TabletColumn& co
     if (tablet_index) {
         opts.need_bloom_filter = true;
         opts.is_ngram_bf_index = true;
-        opts.gram_size = tablet_index->get_gram_size();
-        opts.gram_bf_size = tablet_index->get_gram_bf_size();
+        //narrow convert from int32_t to uint8_t and uint16_t which is dangerous
+        auto gram_size = tablet_index->get_gram_size();
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/segment_writer.cpp:196]: vertcial_segment_writer::_create_column_writer()

**Review by qidaye — APPROVED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**